src/selectize.js

```diff
@@ -773,7 +773,13 @@ $.extend(Selectize.prototype, {
 			if (typeof value !== 'undefined') {
 				self.lastQuery = null;
 				self.setTextboxValue('');
-				self.addItem(value);
+				if (self.caretPos === self.items.length) {
+					self.addItem(value);
```

Comment:
Shouldn't this logic be inside `addItem`? This is still adding an item, just fixing the order of how it is added